Addapache

BenediktMKuehne opened the pull request:
This adds the Apache Server as wsgi-server to the frontend.

Some things to think about:

- What would you say is the max file-size a firmwares are allowed to have?(for upload)

- Currently the -D (dev) and the default installation are compatible with each other. keep or change?

BenediktMKuehne committed:
ch

BenediktMKuehne committed:
fix

BenediktMKuehne committed:
fix

BenediktMKuehne committed:
ch

BenediktMKuehne committed:
ch django check

BenediktMKuehne committed:
ch flags

BenediktMKuehne committed:
comment

BenediktMKuehne committed:
ch daphne settings

BenediktMKuehne committed:
fix?

BenediktMKuehne committed:
rm collect

BenediktMKuehne committed:
fix collect

BenediktMKuehne committed:
ch server procs

BenediktMKuehne committed:
try

BenediktMKuehne committed:
ch httpd flags

BenediktMKuehne committed:
ch package

BenediktMKuehne committed:
update pip

BenediktMKuehne committed:
ch dir collect

BenediktMKuehne committed:
ch apache exec user

BenediktMKuehne committed:
ch loc of venv

BenediktMKuehne committed:
ch simplify

BenediktMKuehne committed:
add env load

BenediktMKuehne committed:
ch owner

BenediktMKuehne committed:
rework settings

BenediktMKuehne committed:
yf

BenediktMKuehne committed:
Merge branch 'Addapache' of https://github.com/BenediktMKuehne/embark into Addapache

BenediktMKuehne committed:
ch serverfiles collect

BenediktMKuehne committed:
Merge branch 'Addapache' of https://github.com/BenediktMKuehne/embark into Addapache

BenediktMKuehne committed:
ch install options

BenediktMKuehne committed:
fix

BenediktMKuehne committed:
collcetion

BenediktMKuehne committed:
fix

BenediktMKuehne committed:
ch copy only on deploy

BenediktMKuehne committed:
wp

BenediktMKuehne committed:
ch pipenv install loc

BenediktMKuehne committed:
Merge branches 'Addapache' and 'Addapache' of https://github.com/BenediktMKuehne/embark into Addapache

BenediktMKuehne committed:
opti

BenediktMKuehne committed:
ch coding rules

BenediktMKuehne committed:
ch daphne stuff

BenediktMKuehne committed:
ch dev settings

BenediktMKuehne committed:
add emba root setting

BenediktMKuehne committed:
Merge branch 'Addapache' of https://github.com/BenediktMKuehne/embark into Addapache

BenediktMKuehne committed:
ch

BenediktMKuehne committed:
working apache config

BenediktMKuehne committed:
add url for emba_logs

BenediktMKuehne committed:
adding flags for pruning

m-1-k-3 commented:
* What would you say is the max file-size a firmwares are allowed to have?(for upload)

I think it should be defaulted to 250MB max and adjustable easily via configuration.

* Currently the -D (dev) and the default installation are compatible with each other. keep or change?

Is there a reason to not keep?

BenediktMKuehne commented:
> * Currently the -D (dev) and the default installation are compatible with each other. keep or change?
> 
> Is there a reason to not keep?

For debugging of the wsgi and the apache configs(ssl etc) they would have to become incompatible.
But personally I would keep the dev installation as-is.

p4cx commented:
> I think it should be defaulted to 250MB max and adjustable easily via configuration.

I especially like this idea. I think we should add all configurations (which are relevant, after EMBArk is running) to a single config file and then the administrator acc can set these in his settings inside of the webui.

BenediktMKuehne committed:
csrf enable

BenediktMKuehne committed:
ch pruning

BenediktMKuehne committed:
add config file for apache

BenediktMKuehne committed:
linter fixes

BenediktMKuehne committed:
ch docker container setup

BenediktMKuehne committed:
add save dir for logs

BenediktMKuehne committed:
ch kill cmd

BenediktMKuehne committed:
add empty .env

BenediktMKuehne committed:
add ex_suc print

BenediktMKuehne committed:
ch emba log env-var

BenediktMKuehne committed:
fix stuff

BenediktMKuehne committed:
ch deps

BenediktMKuehne committed:
container ch

BenediktMKuehne committed:
l63 fix

BenediktMKuehne committed:
fix

BenediktMKuehne committed:
fix .env prob

BenediktMKuehne committed:
ch netmasks

BenediktMKuehne committed:
ch print text

BenediktMKuehne committed:
fix addr

BenediktMKuehne committed:
set debug for testing

BenediktMKuehne committed:
ch hostname

BenediktMKuehne committed:
try this conf

BenediktMKuehne committed:
exposed but ok

BenediktMKuehne committed:
ch inst process

BenediktMKuehne committed:
ch install steps

BenediktMKuehne committed:
fix err

BenediktMKuehne committed:
add note

BenediktMKuehne committed:
Merge branch 'main' into Addapache

BenediktMKuehne committed:
Merge branch 'main' into Addapache

BenediktMKuehne committed:
ch docker reset

BenediktMKuehne committed:
fixes

BenediktMKuehne committed:
fix

BenediktMKuehne committed:
ch loc

BenediktMKuehne commented:
**Short description of what I've done in this PR:**

- changed some locations of files
- overhauled the installation process (-d is default , -F is Dev etc)
- Added Apache standalone with user-acc and its own directory
- split settings for deploy and development
- enabled docker-linting (except Dockerfile)
- changes that will be needed for docker-deployment
- uninstall option (not 100% yet)
- fixed some csrf stuff

BenediktMKuehne committed:
fix pip

BenediktMKuehne committed:
fixes

BenediktMKuehne committed:
env recreate

BenediktMKuehne committed:
ch export

BenediktMKuehne committed:
dev setup ch

BenediktMKuehne committed:
update versions

BenediktMKuehne committed:
set debug

BenediktMKuehne commented:
`-F` option works now as expected

BenediktMKuehne committed:
fix emba log dir setting

BenediktMKuehne committed:
update pipenv

BenediktMKuehne committed:
ch port

BenediktMKuehne committed:
add note

BenediktMKuehne committed:
fix upload dir

BenediktMKuehne committed:
add color

BenediktMKuehne committed:
fix lint error

BenediktMKuehne committed:
add bandit

BenediktMKuehne committed:
fix venv location prob

BenediktMKuehne committed:
ch use in-proj venv

BenediktMKuehne committed:
fix log error

BenediktMKuehne committed:
rm recom

m-1-k-3 commented on README.md:
```diff
@@ -32,9 +32,12 @@ Furthermore *EMBArk* improves the data provision by aggregating the various scan
 
 ## Start Server
-To start the EMBArk-Server simply run `$ sudo ./embark/run-server.sh`.
+To start the EMBArk-Server simply run `$ sudo pipenv run ./run-server.sh`.
```
in the installation script you are refering to
`Ready to use $sudo ./run-server.sh `

m-1-k-3 commented on README.md:
```diff
@@ -32,9 +32,12 @@ Furthermore *EMBArk* improves the data provision by aggregating the various scan
 This starts the http-Server on 0.0.0.0:80
 
+## Developer
+For devs we recommend simply using: `sudo ./installer.sh -D ` and the `./dev-tools/debug-server-start.sh` script.
```
For developers ...

m-1-k-3 commented on installer.sh:
```diff
@@ -16,35 +16,29 @@
+  echo -e "$CYAN-d$NC         EMBArk default installation"
+  echo -e "$CYAN-F$NC         Installation of EMBArk for developers"
+  echo -e "$CYAN-e$NC         Install EMBA only"
+  echo -e "$CYAN-D$NC         Install for Docker deployment"
```
Developer installation mode

BenediktMKuehne replied:
changed to `-F`, same as EMBA

BenediktMKuehne replied:
`-D` i planed for docker.
which isn't ready yet

m-1-k-3 commented on installer.sh:
```diff
@@ -54,28 +48,33 @@ install_emba() {
+  docker container prune -f --filter "label=flag"
+
+
+  #while docker images | grep -qE "\<none\>"; do
```
if not needed please remove

m-1-k-3 commented:
please finish it up with some final changes ... Currently some tests are running in my environment. If everything runs smooth we can merge afterwards.

Good job!

BenediktMKuehne committed:
rm note

BenediktMKuehne committed:
update options

BenediktMKuehne committed:
Merge branch 'main' into Addapache

BenediktMKuehne committed:
Update dev.py

simplify emba log dir

BenediktMKuehne committed:
Update dev.py

fix emba logging

m-1-k-3 commented:
real a huge step in the right direction. Thank you
